### PR TITLE
Fixed autoPublish bugfix

### PR DESCRIPTION
**Pull request by idanstark42**

### CODE REVIEW

File: client/factories/score.js [hunk 1]
```diff
@@ -122,8 +122,12 @@ function Score (tournament, challenge, logger) {
         matchId: score.matchId,
         round: score.round,
         stage: score.stage,
-        noShow: !!score.noShow,
-        public: !!score.public
+        noShow: !!score.noShow
```
Comment (2roy999) [inline client/factories/score.js]: use `Boolean(score.noShow)` much  clearer

File: client/factories/score.js [hunk 2]
```diff
@@ -122,8 +122,12 @@ function Score (tournament, challenge, logger) {
...
+      }
+
+      // Don't add the `public` field if it isn't there already, as to not overrun to autoPublish in the server.
+      if (score.public) {
```
Comment (2roy999) [inline client/factories/score.js]: if `score.public === true` then the command in the block is redundant

File: client/components/scoresheet/metadata_inputs/metadata_inputs_controller.js [hunk 3]
```diff
@@ -98,13 +98,14 @@ class MetadataInputsController {
           return this.loadMatchOptions().then(() => this.autoselectMatch())
         }
       })
+      .catch(error => this.logger.error(error))
       .then(() => { this.autoselecting = false })
   }
 
   autoselectTeam () {
+    if (!this.refIdentity.table) return Promise.resolve()
```
Comment (2roy999) [inline client/components/scoresheet/metadata_inputs/metadata_inputs_controller.js]: Not sure why the eslint isn't complaining on that one but give each if-statement it's own block, even if it's include 1-line

**Comment by idanstark42** [comment]
Did you `yarn build`?

**Comment by alanggreen** [comment]
Probably not 😒

**Comment by alanggreen** [comment]
> Probably not 😒
> […](#)
> On Mon, Sep 23, 2019, 11:49 AM idanstark42 ***@***.***> wrote: Did you yarn build? — 

Did `yarn build` and tried again. Buttons OK now but still not auto-publishing
image
I verified that `autoPubish` is `true` in the `module.yml` file

**Comment by idanstark42** [comment]
Can you verify that the folks l changes here exist in your local version? Specifically the `factories/score.js` changes?

**Comment by alanggreen** [comment]
Will do in 10 minutes

**Comment by alanggreen** [comment]
> Can you verify that the folks l changes here exist in your local version? Specifically the `factories/score.js` changes?

I do have `score.js`
image
I checked the 2nd file too. Also up to date

**Comment by alanggreen** [comment]
Good work. Public is enabled now. One small request - please take a look at the error log `E [ main  ] client: [object Object]`
```
[ main  ] GET /cd6c777f1945164224dee082abaea03a.woff2 - 200 from  in 3.900 ms
D [ main  ] GET /0ab54153eeeca0ce03978cc463b257f7.woff2 - 200 from  in 6.135 ms
D [ main  ] GET /challenge/2019_US - 304 from  in 11.871 ms
D [ main  ] GET /team/10/matches - 304 from  in 0.737 ms
D [ main  ] GET /team/all - 304 from  in 0.822 ms
D [ main  ] GET /table/all - 304 from  in 1.936 ms
D [ main  ] GET /team/16/matches - 304 from  in 0.817 ms
D [ main  ] GET /team/5/matches - 304 from  in 0.599 ms
I [ main  ] client: Connected to mhub
D [ main  ] GET /team/6/matches - 200 from  in 0.383 ms
W [ main  ] GET /match/upcoming/table/1/2 - 404 from  in 1.351 ms
E [ main  ] client: [object Object]
I [ main  ] Saving score for team 6 on practice stage with 135 pts.
D [ main  ] POST /scores/create - 201 from  in 11.338 ms
I [ main  ] Connected to mhub
I [ main  ] client: Score saved - #6 in practice #1: 135
W [ main  ] GET /match/upcoming/table/1/2 - 404 from  in 0.279 ms
E [ main  ] client: [object Object]
```

**Pull request merged**